components/search-form.tsx

```diff
@@ -2,11 +2,13 @@
 /* eslint-disable @typescript-eslint/no-unsafe-member-access */
 /* eslint-disable @typescript-eslint/no-unsafe-assignment */
 /* eslint-disable @typescript-eslint/no-explicit-any */
-import { ChangeEvent, useState } from "react";
+import React, { ChangeEvent, useCallback, useEffect, useState } from "react";
 
 import { PrimaryButton, SecondaryButton } from "./ui/button";
 import { Select } from "./ui/select";
 
+import { debounce } from "lodash";
+
```

Comment:
Hmmm, `lodash` 😴

Can we just create our own debounce? It's not that hard.
Once we add lodash, it's easy to abuse the usage

Reply:
I changed it to use `ts-debounce` instead, could you check if it's okay? Thank you

Reply:
https://bundlephobia.com/package/ts-debounce@3.0.0